pkg/scheduler/algorithm/predicates/csi_volume_predicate.go

```diff
@@ -51,51 +54,47 @@ func NewCSIMaxVolumeLimitPredicate(
 func (c *CSIMaxVolumeLimitChecker) attachableLimitPredicate(
 	pod *v1.Pod, meta PredicateMetadata, nodeInfo *schedulernodeinfo.NodeInfo) (bool, []PredicateFailureReason, error) {
 
-	// if feature gate is disable we return
 	if !utilfeature.DefaultFeatureGate.Enabled(features.AttachVolumeLimit) {
 		return true, nil, nil
 	}
-	// If a pod doesn't have any volume attached to it, the predicate will always be true.
-	// Thus we make a fast path for it, to avoid unnecessary computations in this case.
-	if len(pod.Spec.Volumes) == 0 {
-		return true, nil, nil
-	}
 
-	nodeVolumeLimits := nodeInfo.VolumeLimits()
-
-	// if node does not have volume limits this predicate should exit
-	if len(nodeVolumeLimits) == 0 {
+	// If the new pod doesn't have any volume attached to it, the predicate will always be true
+	if len(pod.Spec.Volumes) == 0 {
```

Comment:
Could you please check this condition before checking the feature gate? It appears that checking the feature gate is not very fast. So, it is better to check volumes length before that. Alternatively, you can combine the two if statements:

```
if len(pod.Spec.Volumes) == 0 || !utilfeature.DefaultFeatureGate.Enabled... {
    return ...
}
```

Reply:
Done